### PR TITLE
smoke-b2c- agent assist

### PR DESCRIPTION
**Pull request by khushal-rathore-metacube**
https://spryker.atlassian.net/browse/CC-21836

- Impersonate / unimpersonate as a customer
- Place an order for a customer

### CODE REVIEW

File: resources/steps/agent_assist_steps.robot [hunk 1]
```diff
@@ -40,3 +40,5 @@ Yves: As an Agent login under the customer:
     Click    xpath=//ul[@data-qa='component customer-list']/li[@data-value='${searchQuery}']
     Click    ${agent_confirm_login_button}
 
+Yves: End customer assistance
```
Comment (AntonShopin) [inline resources/steps/agent_assist_steps.robot:43]: Remane to `Yves: end customer assistance`

File: tests/smoke/smoke_b2c.robot [hunk 2]
```diff
@@ -406,13 +406,25 @@ Agent_Assist
...
+    Yves: perform search by customer:    ${yves_second_user_first_name}
+    Yves: agent widget contains:    ${yves_second_user_email}
+    Yves: As an Agent login under the customer:    ${yves_second_user_email}
+    Yves: perform search by:    020
+    Yves: product with name in the catalog should have price:    Sony Cyber-shot DSC-W830    €105.80
+    Yves: go to PDP of the product with sku:    020
+    Yves: product price on the PDP should be:    €105.80
+    Yves: add product to the shopping cart
+    Yves: go to b2c shopping cart    
+     Yves: click on the 'Checkout' button in the shopping cart
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2c.robot:418]: please fix the alignment

File: tests/smoke/smoke_b2c.robot [hunk 3]
```diff
@@ -406,13 +406,25 @@ Agent_Assist
...
+      Yves: fill in the following new shipping address:
+    ...    || firstName               |     lastName                    |    street        |    houseNumber      |    city          |    postCode    |    phone        ||
+    ...     || ${yves_second_user_first_name}    |     ${yves_second_user_last_name}         |    ${random}              |    ${random}              |    Berlin${random}        |   ${random}              |    ${random}       ||
+    Yves: submit form on the checkout
+    Yves: select the following shipping method on the checkout and go next:     Standard: €4.90
+    Yves: select the following payment method on the checkout and go next:    Invoice
+    Yves: accept the terms and conditions:    true
+    Yves: 'submit the order' on the summary page
+    Yves: End customer assistance
```
Comment (AntonShopin) [inline tests/smoke/smoke_b2c.robot:427]: You also need to test that in Zed or Yves the new order has been created by the customer. 
So we log in under Bill using the Agent assist feature and place an order. In the end, we need to check that Bill has a new order

**Review by AntonShopin — CHANGES_REQUESTED** [review]

File: resources/steps/agent_assist_steps.robot (hunk 1 above)
Comment (khushal-rathore-metacube) [reply resources/steps/agent_assist_steps.robot:43]: Done

File: tests/smoke/smoke_b2c.robot (hunk 2 above)
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:418]: Done

File: resources/steps/agent_assist_steps.robot [hunk 4]
```diff
@@ -26,17 +27,20 @@ Yves: perform search by customer:
...
 
-
-
 Yves: agent widget contains:
     [Arguments]    ${searchQuery}
     Wait Until Element Is Visible    xpath=//ul[@data-qa='component customer-list']/li[@data-value='${searchQuery}']
     Page Should Contain Element    xpath=//ul[@data-qa='component customer-list']/li[@data-value='${searchQuery}']
 
-Yves: As an Agent login under the customer:
+Yves: as an agent login under the customer:
```
Comment (AntonShopin) [inline resources/steps/agent_assist_steps.robot:35]: thank you for formatting it in the right way. 
Could you please replace in all other tests/files this keyword so we don't have a situation, when part of the tests use `Yves: As an Agent login under the customer:` and only 2 `Yves: as an agent login under the customer:`
It should be really simple and quick with replace function in IDE

**Review by AntonShopin — COMMENTED** [review]
Good job. 2 minor changes and we are done here

File: tests/smoke/smoke_b2c.robot [hunk 5]
```diff
@@ -406,13 +406,30 @@ Agent_Assist
...
     Yves: login on Yves with provided credentials:    agent+${random}@spryker.com    change${random}
     Yves: header contains/doesn't contain:    true    ${customerSearchWidget}
-    Yves: perform search by customer:    ${yves_user_first_name}
-    Yves: agent widget contains:    ${yves_user_email}
-    Yves: As an Agent login under the customer:    ${yves_user_email}
-    Yves: perform search by:    031
-    Yves: product with name in the catalog should have price:    Canon PowerShot G9 X    €400.24
-    Yves: go to PDP of the product with sku:    031
-    Yves: product price on the PDP should be:    €400.24
+    Yves: perform search by customer:    ${yves_second_user_first_name}
```
[the remark this answers is not in the document]
Comment (khushal-rathore-metacube) [reply tests/smoke/smoke_b2c.robot:409]: Done

File: resources/steps/agent_assist_steps.robot (hunk 4 above)
Comment (khushal-rathore-metacube) [reply resources/steps/agent_assist_steps.robot:35]: Done

**Review by AntonShopin — APPROVED** [review]

**Pull request merged**